dashboard/lti_new.py

```diff
@@ -0,0 +1,196 @@
+    return tool_config
+
+
+def get_jwks(request):
+    try:
+        tool_conf = get_tool_conf()
+    except (OSError, Exception) as e:
+        error_message = {
+            'lti_error_message': f'Fetching JWKS failed due to {e}'
+        }
+        return JsonResponse(error_message, status=500)
+    return JsonResponse(tool_conf.get_jwks())
+
+
+def check_if_instructor(roles, username, course_id):
+    if INSTRUCTOR in roles:
+        logger.info(f'user {username} is Instructor in the course {course_id}')
+        return True
+    return False
+
+
+def course_user_roles(roles, username):
+    user_membership_roles = set([role for role in roles if role.find(COURSE_MEMBERSHIP) == 0])
+    if not user_membership_roles:
+        logger.info(f'User {username} does not have course membership roles, must be admin {roles}')
+    elif INSTRUCTOR in user_membership_roles and TA in user_membership_roles:
+        logger.info(f'User {username} is a {TA} in the course')
+        user_membership_roles.remove(INSTRUCTOR)
+    return user_membership_roles
+
+
+def short_user_role_list(roles):
+    short_role = []
+    for role in roles:
+        short_role.append(role.split('#')[1])
+    return short_role
+
+
+def extracting_launch_variables_for_tool_use(request, message_launch):
+    launch_id = message_launch.get_launch_id()
+    launch_data = message_launch.get_launch_data()
+    logger.debug(f'lti launch data {launch_data}')
+    custom_params = launch_data['https://purl.imsglobal.org/spec/lti/claim/custom']
+    logger.debug(f'lti_custom_param {custom_params}')
+    if not custom_params:
+        raise Exception(
+            f'You need have custom parameters configured on your LTI Launch. Please see the LTI installation guide on the Github Wiki for more information.'
+        )
+    course_name = launch_data['https://purl.imsglobal.org/spec/lti/claim/context']['title']
+    roles = launch_data['https://purl.imsglobal.org/spec/lti/claim/roles']
+    username = custom_params['user_username']
+    course_id = custom_params['canvas_course_id']
+    email = launch_data['email']
+    first_name = launch_data['given_name']
+    last_name = launch_data['family_name']
+    user_img = launch_data['picture']
+
+    # Add user to DB if not there; avoids Django redirection to login page
+    try:
+        user_obj = User.objects.get(username=username)
+    except User.DoesNotExist:
+        password = ''.join(random.sample(string.ascii_letters, settings.RANDOM_PASSWORD_DEFAULT_LENGTH))
+        user_obj = User.objects.create_user(username=username, email=email, password=password, first_name=first_name,
+                                            last_name=last_name)
+    user_obj.backend = 'django.contrib.auth.backends.ModelBackend'
+    django.contrib.auth.login(request, user_obj)
+    user_roles = course_user_roles(roles, username)
+    is_instructor = check_if_instructor(user_roles, username, course_id)
+    short_roles_list = short_user_role_list(user_roles)
```

Comment:
This goes along with my [other suggestion](https://github.com/tl-its-umich-edu/my-learning-analytics/pull/982/files#r463757435):

```suggestion
```